src/XRP/model/xrp-payment.ts

```diff
@@ -18,20 +19,23 @@ export default class XrpPayment {
    * @return an XrpPayment with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L224
    */
-  public static from(
-    payment: Payment,
-    xrplNetwork: XrplNetwork,
-  ): XrpPayment | undefined {
+  public static from(payment: Payment, xrplNetwork: XrplNetwork): XrpPayment {
     const paymentAmountValue = payment.getAmount()?.getValue()
-    const amount =
-      paymentAmountValue && XrpCurrencyAmount.from(paymentAmountValue)
-    if (!amount) {
-      return undefined // amount is required
+    if (!paymentAmountValue) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Payment protobuf is missing required `amount` field.',
+      )
     }
+    const amount = XrpCurrencyAmount.from(paymentAmountValue)
+    // For non-XRP amounts, the nested field names in `amount` are lower-case.
```

Comment:
I think we can omit checks for all of these lower-case situations.  i'm not sure if there are any similar checks implemented in `XrpCurrencyAmount`, but if we one day decide to add them they'd probably go in that class anyway.

Reply:
`XrpCurrencyAmount` and `XrpIssuedCurrency` don't specify any checks like this that need to be implemented. 

I'll just leave in the comments for now, just so it's noted, without a TODO or anything.